From fc6934c93ced7ce2d3085ec766b0bf5273bc4db7 Mon Sep 17 00:00:00 2001
From: Eric Bentley <eric.bentley@lairdtech.com>
Date: Fri, 27 Dec 2013 12:27:07 -0500
Subject: [PATCH] clear saved  APNAME and APIP on disconnect

The APNAME and APIP are now collected from the various IEs from the
association response.  The SDK can now pull this information for
passing to it's status command.

The stored APNAME and APIP need to be cleared on disconnect so the
SDK does not get stale info

BZ 2251
---
 drivers/net/wireless/ath/ath6kl/wmi.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath6kl/wmi.c b/drivers/net/wireless/ath/ath6kl/wmi.c
index eef1ee2..128344a 100644
--- a/drivers/net/wireless/ath/ath6kl/wmi.c
+++ b/drivers/net/wireless/ath/ath6kl/wmi.c
@@ -1003,6 +1003,9 @@ static int ath6kl_wmi_disconnect_event_rx(struct wmi *wmi, u8 *datap, int len,
 	struct wmi_disconnect_event *ev;
 	wmi->traffic_class = 100;
 
+	__lrd_set_AP_Name(wmi, NULL); // clear existing
+	__lrd_set_AP_IP(wmi, NULL);  // clear existing
+
 	if (len < sizeof(struct wmi_disconnect_event))
 		return -EINVAL;
 
-- 
1.7.9.5

